src/DSharp.Compiler/Compiler/MetadataBuilder.cs

```diff
@@ -1276,8 +1276,16 @@ private void BuildTypeInheritance(ClassSymbol classSymbol)
 
                 foreach (NameNode node in customTypeNode.BaseTypes)
                 {
+                    string nodeName = node.Name;
+
+                    if (node is GenericNameNode genericNameNode)
+                    {
+                        nodeName += $"`{genericNameNode.TypeArguments.Count}";
```

Comment:
Could we not push this down into the GenericNameNode and have a new property like `GenericName` to expose this?

Reply:
lets leave this for the future

Reply:
Sure, It's not urgent. I can always refactor in any generics work.